src/web/app/pages-logs/logs-page.component.ts

```diff
@@ -0,0 +1,200 @@
+import { LogService } from '../../services/log.service';
+import { StatusMessageService } from '../../services/status-message.service';
+import { LOCAL_DATE_TIME_FORMAT, TimeResolvingResult, TimezoneService } from '../../services/timezone.service';
+import { ApiConst } from '../../types/api-const';
+import { GeneralLogEntry, GeneralLogs, Type } from '../../types/api-output';
+import { LogsTableRowModel } from '../components/logs-table/logs-table-model';
+import { DateFormat } from '../components/session-edit-form/session-edit-form-model';
+import { TimeFormat } from '../components/session-edit-form/time-picker/time-picker.component';
+import { ErrorMessageOutput } from '../error-message-output';
+
+/**
+ * Model for searching of logs.
+ */
+interface SearchLogsFormModel {
+  logsSeverity: Set<string>;
+  logsDateFrom: DateFormat;
+  logsDateTo: DateFormat;
+  logsTimeFrom: TimeFormat;
+  logsTimeTo: TimeFormat;
+}
+
+/**
+ * Query parameters for HTTP request
+ */
+interface QueryParams {
+  searchFrom: string;
+  searchUntil: string;
+  severities: string;
+  nextPageToken?: string;
+}
+
+/**
+ * Admin and maintainer logs page.
+ */
+@Component({
+  selector: 'tm-logs-page',
+  templateUrl: './logs-page.component.html',
+  styleUrls: ['./logs-page.component.scss'],
+})
+export class LogsPageComponent implements OnInit {
+  LOGS_RETENTION_PERIOD_IN_DAYS: number = ApiConst.LOGS_RETENTION_PERIOD;
+  LOGS_RETENTION_PERIOD_IN_MILLISECONDS: number = this.LOGS_RETENTION_PERIOD_IN_DAYS * 24 * 60 * 60 * 1000;
+  SEVERITIES: string[] = ['INFO', 'WARNING', 'ERROR'];
+
+  formModel: SearchLogsFormModel = {
+    logsSeverity: new Set(),
+    logsDateFrom: { year: 0, month: 0, day: 0 },
+    logsTimeFrom: { hour: 0, minute: 0 },
+    logsDateTo: { year: 0, month: 0, day: 0 },
+    logsTimeTo: { hour: 0, minute: 0 },
+  };
+  previousQueryParams: QueryParams = { searchFrom: '', searchUntil: '', severities: '' };
+  dateToday: DateFormat = { year: 0, month: 0, day: 0 };
+  earliestSearchDate: DateFormat = { year: 0, month: 0, day: 0 };
+  searchResults: LogsTableRowModel[] = [];
+  isLoading: boolean = false;
+  isSearching: boolean = false;
+  hasResult: boolean = false;
+  nextPageToken: string = '';
+
+  constructor(private logService: LogService,
+    private timezoneService: TimezoneService,
+    private statusMessageService: StatusMessageService) { }
+
+  ngOnInit(): void {
+    const today: Date = new Date();
+    this.dateToday.year = today.getFullYear();
+    this.dateToday.month = today.getMonth() + 1;
+    this.dateToday.day = today.getDate();
+
+    const earliestSearchDate: Date = new Date(Date.now() - this.LOGS_RETENTION_PERIOD_IN_MILLISECONDS);
+    this.earliestSearchDate.year = earliestSearchDate.getFullYear();
+    this.earliestSearchDate.month = earliestSearchDate.getMonth() + 1;
+    this.earliestSearchDate.day = earliestSearchDate.getDate();
+
+    this.formModel.logsDateFrom = { ...this.dateToday, day: today.getDate() - 1 };
```

Comment:
Will this hit the same problem as #11246?

Reply:
Yes. Will fix this issue